src/west/app/project.py

```diff
@@ -421,14 +419,19 @@ def delay(func, project):
             # as SHAs, unless they are specifically requested, and then
             # ensures they are only computed once.
             try:
-                if (isinstance(project, ManifestProject) and not
-                        args.manifest_path_from_yaml):
+                if isinstance(project, ManifestProject):
                     # Special-case the manifest repository while it's
                     # still showing up in the 'projects' list. Yet
                     # more evidence we should tackle #327.
-                    path = self.config.get('manifest.path')
-                    apath = abspath(os.path.join(self.topdir, path))
-                    ppath = Path(apath).as_posix()
+                    if args.manifest_path_from_yaml:
+                        path = self.manifest.yaml_path
```

Comment:
similar to the comment on `get_option('manifest.path')`.

Shouldn't `project.path` be identical to `yaml_path` when loading from the yaml and not topdir ?
https://github.com/zephyrproject-rtos/west/blob/9dbc5cdf13374daa443d4e6b36bc6114d5356fa7/src/west/manifest.py#L1872-L1885

and
https://github.com/zephyrproject-rtos/west/blob/9dbc5cdf13374daa443d4e6b36bc6114d5356fa7/src/west/manifest.py#L1788-L1789

I share the same concern as described in the `get_option('manifest.path')` comment.

If there is a use-case that I have not noticed, then please provide extra info, but i'm concerned that special treatment of the path from `ManifestProject` bears a risk of unexpected behavior to any usages of `manifest.get_projets()` and confusion (and problem that already exists today, and must be addressed, just hoping not to increase confusion).

For example, Zephyr uses `get_projects()` directly instead of `west list`.
https://github.com/zephyrproject-rtos/zephyr/blob/a4d4384447710e9885c0248385db0d5a5d197d50/scripts/zephyr_module.py#L377-L380

Reply:
No, they are different in this case.

For example, with this diff to `zephyr/west.yml`:

```
diff --git a/west.yml b/west.yml
index 35cb064ce3..453b2366aa 100644
--- a/west.yml
+++ b/west.yml
@@ -245,6 +245,6 @@ manifest:
     - -ci
 
   self:
-    path: zephyr
+    path: torsten's-house
     west-commands: scripts/west-commands.yml
     import: submanifests
```

We have:

```
$ west list --manifest-path-from-yaml -f {path} manifest
torsten's-house
```

But the `ManifestProject.path` attribute is still `zephyr` because `self._config_path` is `zephyr` when we have a topdir that has `manifest.path=zephyr`.

Reply:
Ah right, this is also the where you might do a `west init -l` on a manifest repo that has a path different from the one in the manifest, but you want `west list` to print the `self::path` from that yaml, and not the actual in use.